### PR TITLE
internal object communication 

### PR DESCRIPTION
**Pull request by vrawatnec**
Comments incorporated.

### CODE REVIEW

File: common/config/config.go [hunk 1]
```diff
@@ -107,6 +112,26 @@ func (c *Config) GetDiscoveryURL() string {
...
+                port = c.Designer.HTTPSPort
+        }
+
+        hostip := c.InternalIP
+        if c.CoreSerivceIP != "" {
+                hostip = c.CoreSerivceIP
+        }
+
+        if c.Discovery.HostIP != "" {
+                hostip = c.Designer.HostIP
```
Comment (naveennec) [inline common/config/config.go:129]: looking logic is wrong. please check it.

File: designer/main.js [hunk 2]
```diff
@@ -141,9 +141,10 @@ app.post('/intent', jsonParser, function(req, res) {
 */
 
 function getResult(filterBy, objList) {
+    console.log("objlist is ---- ",objList);
     var arr = [];
     for(var i in objList){
-            if (Object.keys(objList[i].attribute).length === 0) {
+            if (objList[i].hasOwnProperty("attribute") && Object.keys(objList[i].attribute).length === 0) {
                 console.log("aaaa",objList[i]);
```
Comment (naveennec) [inline designer/main.js:148]: need to remove the print statements

File: designer/main.js [hunk 3]
```diff
@@ -186,6 +187,10 @@ app.post('/internal/getContext', jsonParser, async function (req, res) {
     res.send({data:getResult(queryContext.internalType,dgraphOp.contextElements)})
 });
 
+app.post('/masterNotify', jsonParser, async function (req, res) {
```
Comment (naveennec) [inline designer/main.js:190]: need to write the comment of this logic.

File: designer/main.js [hunk 4]
```diff
@@ -241,6 +246,22 @@ function handleNotify(req, ctxObjects, res) {
     }
 }
 
+async function dgraphSendToMaster(){
```
Comment (naveennec) [inline designer/main.js:249]: Missing comment section, what is the motive of this function.

File: designer/public/js/anomaly.js [hunk 5]
```diff
@@ -105,25 +112,47 @@ $(function() {
...
+                    geoscope = result[0].geoscope;
+                }
+                
             }
         }).catch(function(error) {
             console.log(error);
             console.log('failed to query context');
         });
     }
+    // function checkIntent() {
```
Comment (naveennec) [inline designer/public/js/anomaly.js:138]: better to write a comment on the top of this commented section why this is commented.

File: designer/public/js/lostchild.js [hunk 6]
```diff
@@ -116,16 +121,21 @@ $(function() {
 
     function checkIntent() {
         var queryReq = {};
-        queryReq.entities = [{ type: 'ServiceIntent', isPattern: true }];
-        queryReq.restriction = { scopes: [{ scopeType: 'stringQuery', scopeValue: 'topology=Topology.child-finder' }] }
-
-        client.queryContext(queryReq).then(function(resultList) {
+       // queryReq.entities = [{ type: 'ServiceIntent', isPattern: true }];
```
Comment (naveennec) [inline designer/public/js/lostchild.js:124]: remove all the irrelevant commented code.

File: master/master.go [hunk 7]
```diff
@@ -112,6 +118,32 @@ func (master *Master) Start(configuration *Config) {
...
+                }
+               //fmt.Println(master.cfg.HTTPS)
+               master.cfg.HTTPS.LoadConfig()
+               client := master.cfg.HTTPS.GetHTTPClient()
+               fmt.Println("==== client =====",client)
+               req2, err := http.NewRequest("POST", master.designerURL+"/masterNotify", bytes.NewBuffer(body))
+               fmt.Println("++++++ req ++++++ and err",req2,err)
+               for {
+                        //resp, err := client.Post(url,"application/json" , bytes.NewBuffer(body))
+                       time.Sleep(5 * time.Second)
```
Comment (naveennec) [inline master/master.go:136]: need to define a variable retryInterval and set the value 5. and use retryInterval variable here.

File: master/master.go [hunk 8]
```diff
@@ -502,12 +535,12 @@ func (master *Master) prefetchDockerImages(image DockerImage) {
...
 
 func (master *Master) handleFogFunctionUpdate(msg json.RawMessage) {
-	//FO.Println(msg)
+	//INFO.Println(msg)
 	var fogfunction = FogFunction{}
         err := json.Unmarshal(msg, &fogfunction)
 	fogfunction.Intent.ID = fogfunction.Id
 	fmt.Println("***** Intent.ID *********",fogfunction.Intent.ID)
-	fmt.Println("********* msg *******",msg, fogfunction)
+	//fmt.Println("********* msg *******",msg, fogfunction)
```
Comment (naveennec) [inline master/master.go:543]: remove the commented code.

File: designer/public/js/utility.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
+(function () {
+
```
Comment (naveennec) [inline designer/public/js/utility.js:2]: add comment in front of each method just to know the motive of method.

**Review by naveennec — CHANGES_REQUESTED** [review]

File: common/config/config.go (hunk 1 above)
Comment (pradumnnec) [reply common/config/config.go:126]: Comment Incorporated

File: master/master.go (hunk 8 above)
Comment (pradumnnec) [reply master/master.go]: Comment Incorporated

**Review by naveennec — APPROVED** [review]

**Pull request merged**